test/helpers/user.ts

```diff
@@ -240,7 +240,11 @@ export const createAndFundNewWallet = async <S extends WalletCurrency>({
   if (wallet instanceof Error) throw wallet
   // Fund new wallet if a non-zero balance is passed
   if (balanceAmount.amount === 0n) return wallet
-  const lnInvoice = await Wallets.addInvoiceForSelf({
+  const addInvoiceFn =
```

Comment:
why did we had not a segregation per wallet before?

I found the function below pretty "hard" to read compare to the previous flow. not an issue really just an impression

Reply:
Yea the changes here (https://github.com/GaloyMoney/galoy/compare/4e85159da684722efece05fff5baba28d9c5a9cf...move-validate-wallet-methods) ended up being quite a lot.

I did it this way because moving the wallet validation into the use-cases meant that the undifferentiated functions should be private to those modules and we should probably only expose the wallet-specific methods. This meant that the undifferentiated methods were no longer available for import into the tests and we'd now have to do explicit differentiation like with here.

Reply:
maybe I'm not a fan of a ternary operator also. the main reason is we're assuming only two base currency, but this is just a temporary thing. eventually we will manage more currencies (not just display currency but base currency) with fiat integration.

Reply:
so a `switch` of explicitly one liner `if` could be not only easier to read but also easier to expand.

Reply:
I'm not sure I'd include a switch here to handle 3+ base currencies. I think at that point we'd have a lot more refactoring to do to pass in amounts + currencies instead of having separate use-cases for each base currency. Including a switch here feels like partial/premature optimization for that.

An ok path to keep readability too could be to revert the last 3 commits and leave the validation call in the queries/mutations. Since this was supposed to be a simple PR to move those methods into the app layer, that could potentially be an ok middle-ground without having to get into refactoring too many other things prematurely.

Reply:
> I'm not sure I'd include a switch here to handle 3+ base currencies. I think at that point we'd have a lot more refactoring to do to pass in "amounts + currencies" instead of having separate use-cases for each base currency. Including a switch here feels like partial/premature optimization for that

this is fair. approved.